src/ImageSharp/Common/Helpers/SimdUtils.Avx2Intrinsics.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace SixLabors.ImageSharp
+{
+    internal static partial class SimdUtils
+    {
+        public static class Avx2Intrinsics
+        {
+            private static ReadOnlySpan<byte> PermuteMaskDeinterleave8x32 => new byte[] { 0, 0, 0, 0, 4, 0, 0, 0, 1, 0, 0, 0, 5, 0, 0, 0, 2, 0, 0, 0, 6, 0, 0, 0, 3, 0, 0, 0, 7, 0, 0, 0 };
```

Comment:
Note for the future. We should add comments to this kind of stuff so I can understand what is actually does! 😆

Reply:
Well, I would be happy if I could place any meaningful comment here, but the truth is that I have no idea what does it do *exactly*.

All I know is that it's a permuatation mask to unshuffle the bytes returned by `PackSignedSaturate` which are in a meaningless order for some reason I not (yet) understand, and haven't taken the time to research it any further. Maybe if @saucecontrol has some more time to clarify the high level concept..

Reply:
The basic idea is that nearly all AVX instructions operate independently on 2 128-bit lanes rather than on the 256-bit register as a whole.  So if you have 4 `Vector256<int>` that contain pixels 0,1 | 2,3 | 4,5 | 6,7, when you narrow and pack them, they end up in 2 registers as 0,2,1,3 | 4,6,5,7.  Then you do that again, and you get 1 register with 0,4,2,6,1,5,3,7.

Permute instructions essentially do a shuffle across lanes, so you give it the order 0,4,1,5,2,6,3,7 to undo the interleaving that happened in the previous steps.  That ROS just has those 4 32-bit integers written in little endian order.

Reply:
> nearly all AVX instructions operate independently on 2 128-bit lanes rather than on the 256-bit register as a whole

Knowing this, the whole thing makes much more sense now.

Reply:
Actually, that's still not a great explanation, because the pixels don't stay together like that... they end up with the channels interleaved as well.  If you try to match up the order with the full pixel numbers like I gave there, it still won't make sense.  I'll draw out a proper diagram when I have a chance.

Reply:
Ha, confused myself by not keeping them in 32-bit chunks.  I updated the numbers above so that they make sense